src/libply/aux/kallsyms.c

```diff
@@ -89,6 +96,74 @@ const struct ksym *ksym_get(struct ksyms *ks, uintptr_t addr)
 		       ks->cache->hdr.n_syms - 1, sizeof(key), ksym_cmp);
 }
 
+static int kfunc_list_cmp(const void *_key, const void *_member)
+{
+	const char * const *key = _key, * const *member = _member;
+
+	return strcmp(*key, *member);
+}
+
+static void kfunc_list_sort(struct kfunc_list *kfuncs)
+{
+	qsort(kfuncs->kfunc, kfuncs->size, sizeof(char *), kfunc_list_cmp);
+}
+
+static int kfunc_list_find(struct kfunc_list *kfuncs, const char *name)
+{
+	return !!bsearch(&name, kfuncs->kfunc, kfuncs->size, sizeof(char *), kfunc_list_cmp);
+}
+
+static int kfunc_list_build(struct kfunc_list *kfuncs)
+{
+	FILE *fp;
+	char line[0x80];
+	char **list = NULL;
+	int size = 0;
+	int count = 0;
+
+	fp = fopen(KFUNC_LIST, "r");
+	if (!fp) {
+		fprintf(stderr, "cannot find %s: %m\n", KFUNC_LIST);
+		return -1;
+	}
+
+	while (fgets(line, sizeof(line), fp)) {
+		char *p;
+		struct ksym *ksym;
+
+		if (count >= size) {
+			size = size ? (size * 2) : 1024;
+			list = realloc(list, size * sizeof(*list));
+			assert(list);
+		}
+
+		p = strtok(line, " \t\n");
+		list[count] = strndup(p, sizeof(ksym->sym) - 1);
```

Comment:
This does not handle kernel module symbols correctly.

Reply:
Right, it'll remove the module names.  But the ksyms in the ply already does that too so I think it's fine for now.

Reply:
Oh, I have misread this. It's OK.

Reply:
Thanks for your review!